src/components/Header.vue

```diff
@@ -7,9 +7,15 @@ import { IItem } from "@/types/treeNodes";
 const route = useRoute();
 const treeStore = useTreeNodes();
 const tree = treeStore.getTree;
+const emit = defineEmits(["toggle-navigation-drawer", "toggle-tree-drawer"]);
```

Comment:
what are "defineEmits" ?

Reply:
https://vuejs.org/api/sfc-script-setup.html#defineprops-defineemits

in Main we have states that responsible for opening and closing drawers. In Header we have buttons that change state in parent component, so we need "defineEmits".